common/src/View/MapViewBase.cpp

```diff
@@ -1043,6 +1048,46 @@ namespace TrenchBroom {
             return nullptr;
         }
         
+        void MapViewBase::OnMergeGroups(wxCommandEvent& event) {
+            MapDocumentSPtr document = lock(m_document);
+            Model::Group* newGroup = findGroupToMergeGroupsInto(document->selectedNodes());
+            ensure(newGroup != nullptr, "newGroup is null");
+            
+            Transaction transaction(document, "Merge Groups");
+            document->mergeSelectedGroupsWithGroup(newGroup);
+        }
+
+        Model::Group* MapViewBase::findGroupToMergeGroupsInto(const Model::NodeCollection& selectedNodes) const {
+            if (!(selectedNodes.hasOnlyGroups() && selectedNodes.groupCount() >= 2)) {
+                return nullptr;
+            }
+            Model::Group* mergeTarget = nullptr;
+            
+            MapDocumentSPtr document = lock(m_document);
+            const Model::Hit& hit = pickResult().query().pickable().type(Model::Group::GroupHit).first();
+            if (hit.isMatch()) {
+                mergeTarget = Model::hitToGroup(hit);
+            }
+            
+            const Model::NodeList& nodes = selectedNodes.nodes();
+            const bool canReparentAll = std::all_of(nodes.begin(), nodes.end(), [&](const auto* node){
+                if (node == mergeTarget) {
+                    return true;
+                } else {
+                    return this->canReparentNode(node, mergeTarget);
+                }
+            });
+            
+            if (canReparentAll) {
+                return mergeTarget;
+            }
+            return nullptr;
+        }
+        
+        bool MapViewBase::canReparentNode(const Model::Node* node, const Model::Node* newParent) const {
```

Comment:
Could this method also be used in other reparenting commands?

Reply:
Yeah, I can re-use this in `MapViewBase::canReparentNodes`.